### PR TITLE
Enhance perf_24x7_hardware test errors

### PR DESCRIPTION
**Pull request by tfauck**
Add more relevant message test if perf not enabled at BMC level
grep -E returns error if ls returns null string

Signed-off-by: Thierry <tfauck@free.fr>

### CODE REVIEW

File: perf/perf_24x7_hardware_counters.py [hunk 1]
```diff
@@ -46,12 +47,14 @@ def setUp(self):
         """
         smm = SoftwareManager()
         detected_distro = distro.detect()
-        if 'ppc' not in process.system_output("uname -p", ignore_status=True).decode("utf-8"):
+        if 'ppc' not in process.system_output("uname -a", ignore_status=True).decode("utf-8"):
```
Comment (PraveenPenguin) [inline perf/perf_24x7_hardware_counters.py]: the earlier condition also woks well so i don't think change is requrie

File: perf/perf_24x7_hardware_counters.py [hunk 2]
```diff
@@ -168,10 +174,9 @@ def test_domain_chip_offset(self):
 
     # Helper functions
     def event_helper(self, event):
-        search_suffix = process.run('ls %s/events |grep -E '
-                                    '%s' % (self.event_sysfs, event),
+        search_suffix = process.run('ls %s/events' % (self.event_sysfs),
                                     ignore_status=True)
-        if search_suffix.stdout.decode("utf-8"):
+        if event in search_suffix.stdout.decode("utf-8"):
```
Comment (PraveenPenguin) [inline perf/perf_24x7_hardware_counters.py]: nice catch

File: perf/perf_24x7_hardware_counters.py (hunk 1 above)
Comment (tfauck) [reply perf/perf_24x7_hardware_counters.py]: No uname -p is not portable so it doesn't work on all distro

Comment (PraveenPenguin) [reply perf/perf_24x7_hardware_counters.py]: ah but i did not see any issue can you please list out which distro you are facing a issue

Comment (tfauck) [reply perf/perf_24x7_hardware_counters.py]: -p sounds supported but may report value unknown - Got that on PowerNV debian system. Going to investigate why.

Comment (tfauck) [reply perf/perf_24x7_hardware_counters.py]: reverting

File: perf/perf_24x7_hardware_counters.py (hunk 2 above)
Comment (tfauck) [reply perf/perf_24x7_hardware_counters.py]: thx

**Comment by tfauck** [comment]
Did wrap both commits in a single commit for perf/perf_24x7...

**Pull request merged**